### PR TITLE
don't fail when the local apidoc cache is broken

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
thanks @wbclark for finding this.

File: tests/test_apypie.py [hunk 1]
```diff
@@ -80,6 +80,26 @@ def test_init_with_existing_cachedir(fixture_dir, requests_mock, tmpdir):
...
 
+def test_init_with_bad_cache(fixture_dir, requests_mock, tmpdir):
+    with fixture_dir.join('dummy.json').open() as read_file:
+        data = json.load(read_file)
+    requests_mock.get('https://api.example.com/apidoc/v1.json', json=data)
+
+    old_environ = os.environ.copy()
+    os.environ['XDG_CACHE_HOME'] = tmpdir.strpath
+
+    json_path = tmpdir.join('apypie/https___api.example.com/v1/deadc0ffee.json')
```
Comment (ekohl) [inline tests/test_apypie.py:91]: Isn't this cleaner?
```suggestion
    json_path = tmpdir.join('apypie', 'https___api.example.com', 'v1', 'deadc0ffee.json')
```

File: tests/test_apypie.py [hunk 2]
```diff
@@ -80,6 +80,26 @@ def test_init_with_existing_cachedir(fixture_dir, requests_mock, tmpdir):
...
+    json_path.ensure(file=True)
+    json_path.write('BAD JSON')
+
+    api = apypie.Api(uri='https://api.example.com')
+    api.apidoc
+    os.environ = old_environ
```
Comment (ekohl) [inline tests/test_apypie.py:97]: Should this be in some `finally` block?

**Review by ekohl — APPROVED** [review]
Makes sense. Is JSONDecodeError a Python 3 thing?

File: tests/test_apypie.py (hunk 2 above)
Comment (mdellweg) [reply tests/test_apypie.py:97]: You could solve this with a fixture.
```python
@pytest.fixture
def save_environ():
    old_environ = os.environ.copy()
    try:
        yield
    finally:
        os.environ = old_environ
```

**Comment by evgeni** [comment]
> Makes sense. Is JSONDecodeError a Python 3 thing?

Yes it is.

File: tests/test_apypie.py (hunk 1 above)
Comment (evgeni) [reply tests/test_apypie.py:91]: given this is also the case in other tests, I'll merge now and refactor all in one go.

File: tests/test_apypie.py (hunk 2 above)
Comment (evgeni) [reply tests/test_apypie.py:97]: I'll try that in another PR (it's a common thing in the tests)

**Pull request merged**